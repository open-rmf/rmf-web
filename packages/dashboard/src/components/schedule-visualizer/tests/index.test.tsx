import React from 'react';
import ScheduleVisualizer from '../index';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

const mockBuildingMap = {
  name: 'name',
  levels: [
    {
      name: 'name',
      places: [],
      elevation: 1,
      images: [],
      doors: [],
      nav_graphs: [],
      wall_graph: { name: '', vertices: [], edges: [], params: [] },
    },
  ],
  lifts: [],
};

it('renders well with the appropriate data', () => {
  URL.createObjectURL = jest.fn();

  const wrapper = shallow(
    <ScheduleVisualizer
      negotiationTrajStore={{}}
      fleets={[]}
      mapFloorLayerSorted={['name']}
      buildingMap={mockBuildingMap}
    />,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
