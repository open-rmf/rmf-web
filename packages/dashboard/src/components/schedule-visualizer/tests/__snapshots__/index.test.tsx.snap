// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders well with the appropriate data 1`] = `
<Map
  attributionControl={false}
  className="makeStyles-map-1"
  crs={
    Object {
      "distance": [Function],
      "getProjectedBounds": [Function],
      "infinite": true,
      "latLngToPoint": [Function],
      "pointToLatLng": [Function],
      "project": [Function],
      "projection": Object {
        "bounds": Object {
          "max": Object {
            "x": 180,
            "y": 90,
          },
          "min": Object {
            "x": -180,
            "y": -90,
          },
        },
        "project": [Function],
        "unproject": [Function],
      },
      "scale": [Function],
      "transformation": Object {
        "_a": 1,
        "_b": 0,
        "_c": -1,
        "_d": 0,
      },
      "unproject": [Function],
      "wrapLatLng": [Function],
      "wrapLatLngBounds": [Function],
      "zoom": [Function],
    }
  }
  id="ScheduleVisualizer"
  maxZoom={8}
  minZoom={0}
  onbaselayerchange={[Function]}
  zoomDelta={0.5}
  zoomSnap={0.5}
>
  <ForwardRef(Leaflet(AttributionControl))
    position="bottomright"
    prefix="OSRC-SG"
  />
  <ForwardRef(Leaflet(LayersControl))
    position="topleft"
  >
    <Overlay
      checked={true}
      name="Robot Trajectories"
    />
    <Overlay
      checked={true}
      data-component="NegotiationTrajCheckbox"
      name="Negotiation Trajectories"
    />
    <Overlay
      checked={true}
      name="Doors"
    />
    <Overlay
      checked={true}
      name="Lifts"
    />
    <Overlay
      checked={true}
      name="Robots"
    />
    <Overlay
      checked={true}
      name="Waypoints"
    />
    <Overlay
      checked={true}
      name="Dispensers"
    />
  </ForwardRef(Leaflet(LayersControl))>
</Map>
`;
